swap/src/cli/cancel_and_refund.rs

```diff
@@ -55,9 +57,31 @@ pub async fn cancel(
 
     tracing::info!(%swap_id, "Manually cancelling swap");
 
-    let (txid, subscription) = match state6.submit_tx_cancel(bitcoin_wallet.as_ref()).await {
-        Ok(txid) => txid,
+    match state6.submit_tx_cancel(bitcoin_wallet.as_ref()).await {
+        Ok((txid, _)) => {
+            let state = BobState::BtcCancelled(state6);
+            db.insert_latest_state(swap_id, state.clone().into())
+                .await?;
+            Ok((txid, state))
+        }
         Err(err) => {
+            if state6
+                .check_for_tx_cancel(bitcoin_wallet.as_ref())
+                .await
+                .is_ok()
+            {
+                // Alice already cancelled, so we are out-of-sync with Alice.
+                let txid = state6
```

Comment:
No need to manually construct the transaction here again. check_for_tx_cancel returns the transaction

Reply:
Sorry, forgot about that part of code.